src/shared/sam/cli/samCliInvoker.ts

```diff
@@ -39,14 +40,18 @@ export function resolveSamCliProcessInvokerContext(
 export class DefaultSamCliProcessInvoker implements SamCliProcessInvoker {
     public constructor(private readonly context: SamCliProcessInvokerContext = resolveSamCliProcessInvokerContext()) {}
 
-    public async invoke(options?: SamCliProcessInvokeOptions): Promise<ChildProcessResult> {
+    public async invoke(
+        options?: SamCliProcessInvokeOptions,
+        channelLogger?: ChannelLogger
```

Comment:
does it make sense for this to be part of `SamCliProcessInvokeOptions` ?

Reply:
yeah i just really hated this syntax:
```
 export function makeRequiredSamCliProcessInvokeOptions(
     options?: SamCliProcessInvokeOptions
-): Required<SamCliProcessInvokeOptions> {
+): Required<Omit<SamCliProcessInvokeOptions, "channelLogger">> {
     options = options || {}
```

Reply:
Oh I see. No strong opinion either way.